### DISCUSSION TITLE
Folder path is blank when adding torrent

### DISCUSSION THREAD

**Discussion by dvds-ok** [Novik/ruTorrent#2546] [Q&A] [answered]
Hi.

On my computer I have **Ubuntu Server 22.04** installed with **ruTorrent 4.1.6** and **Apache2**.

When I try to add any torrent from the ruTorrent interface, the folder path is blank and it is impossible to select no folder, as seen in the image. At the configuration level I have not customized anything.

![ruTorrent AgregarTorrent](https://github.com/Novik/ruTorrent/assets/97465992/7a01dc52-4c0b-4169-8fec-936ec367951e)

I keep the ruTorrent configuration file (/var/www/rutorrent/conf/**config.php**) with the default configuration.
In November 2022, the user **@Oreloth** opened a ticket on the same subject #2406, so I have tried different values for the **$topDirectory** variable, which defaults to '/', with no apparent effect.

I have the _getdir plugin loaded and activated. I have tried version 4.2.0 of ruTorrent with the same results.

Could anyone give me a clue to fix the problem?

**Comment by Oreloth** [comment]
Hello.

It's been a while, but if I remember correctly it was a problem of permission between the account that manages your seedbox and the account that is used for web files. The easiest thing to do is to put the two together in a common sub-folder, so that both services are managed by the same user.

I hope this helps you in your quest to share.

**Comment by dvds-ok** [comment]
Hello.

Thanks for your answer @Oreloth.
Although I don't use seedbox I have checked permissions and I see them correct. It's more like on my old computer I have installed Linux Mint with ruTorrent 3.1.0 and it works perfectly, I have been comparing permissions in different folders and they are identical, but in the old installation you can see the directory tree from ruTorrent and in the new one you don't.
So far I have made no progress in solving my problem. Since I think the problem is not in the permissions, I'm going to check the configuration from the Web server and from Php.

**Reply by Oreloth** [reply]
By "seedbox" I mean "ruTorrent" and by web server I mean "Apache". I'm trying to say my main problem was about ruTorrent and Apache running on differents users, I had to manage to set them both on the same user to solve this. Anyway I wont help for Ubuntu since I dropped Deb OS for Enterprise Grade OS.

**Reply by dvds-ok** [reply]
At first I understood that you had contracted a **paid seedbox**.
I have tried to use a system user with permissions as a Web user. In that case there shouldn't be permissions problems, but it doesn't work either.

**Reply by Oreloth** [reply]
No, of course, I use ruTorrent as "handmade" seedbox.

**Comment by Oreloth** [comment]
Hi.

To be honest, I was tired of Ubuntu and Debian, nothing worked as expected about ruTorrent. So I just switched to **CentOS** (dead now, I'm on AlmaLinux) and everything is doing fine.

**Comment by stickz** [comment]
Hello, checking the configuration of your web server is a good step. I do not have much experience running ruTorrent with apache2. My configuration is Ubuntu 22.04 LTS with nginx. Switching to CentOS/AlmaLinux is probably just using a different configuration of apache.

There is a good installation script made by swizzin, which works well on Ubuntu server 22.04. It's what I'm using currently without issues. I've made multiple contributions and continue to maintain ruTorrent on it. https://github.com/Novik/ruTorrent/discussions/2482

It's not recommended to use ruTorrent v3.10. There are serious issues and many problems with it, including security vulnerabilities.

**Reply by dvds-ok** [reply]
**@stickz**, I've tried various scripts that automate the installation and configuration of rTorrent/ruTorrent. The one you recommended me (**swizzin**) is the only one which works perfectly without having to do additional things and is therefore highly recommended. 
I am very grateful for your suggestion.

**Comment by dvds-ok** [comment, marked as answer]
**@stickz @Oreloth** 
After thinking about this issue a lot, in the end it was also **a permissions issue**. 
All my problems are solved. **Thank you all!!!!**
**Ubuntu Server 22.04**  with **rTorrent**/**ruTorrent 4.2.2** and **Apache2**

**Reply by Oreloth** [reply]
Said it. Enjoy now.
